src/actions/auth.js

```diff
@@ -0,0 +1,51 @@
+import axios from "axios";
+
+export const USER_CREATED = "USER_CREATED";
+
+export const LOGIN_SUCCESS = "LOGIN_SUCCESS";
+
+const baseUrl = "http://localhost:4000";
+
+function signUpSuccess() {
+  return { type: USER_CREATED };
+}
+
+export function signUp(username, name, surname, email, password) {
+  return async function(dispatch, getState) {
+    try {
+      const response = await axios.post(`${baseUrl}/signup`, {
+        username: username,
+        name: name,
+        surname: surname,
+        email: email,
+        password: password
```

Comment:
if you are using the same `key` name as the variable name then you don't need to repeat it, this works:
```
 await axios.post(`${baseUrl}/signup`, { username, name, surname, email, password })
```